### ISSUE TITLE
Reducing the horizontal width/spacing between files to utilize more space

### ISSUE THREAD

**Issue by enriquezmark36** [linuxmint/nemo#802]
The spacing between the file and folders in a regular window, which is supposedly more than enough to fit an icon of a file or folder, is quite large and might be a bit annoying if one have to look to a lots of files with thumbnails.**(Illus.1)** Zooming down  doesn't help since the gap is still quite large by comparison plus the icons shrink down too.
The current one looks neat but wouldn't it be better if it can be reduced or changed via the code or the settings?

**Illus.1**
![screenshot from 2014-12-21 22 05 37](https://cloud.githubusercontent.com/assets/7193373/5518352/20806072-895e-11e4-924f-431de1432494.png)

**Comment by joe1970** [comment]
+1, I agree, this behaviorto leave that much space unused is strange. There is however the compact layout for the icon view. It can be toggled in "arrange Icons" in the context menue. It seems to pack icons next to each other as long as there is still space in a line left. The result is a nicely packed view, but icons are not arranged in a grid any more.

**Comment by ManIVIctorious** [comment]
@joe1970 i know about this one, but that destroys the alignment of the following row. I'd like to see a better alignment or an option to change the spacing manually (doesn't even have to be exposed, a gsettings call would be enough for me)

**Comment by enriquezmark36** [comment]
@ManIVIctorious, @joe1970, After studying the code for about 3 days there seem to be a way to reduce the width and make use of the space like this: 
![nemosend](https://cloud.githubusercontent.com/assets/7193373/12535061/01c59f98-c2b1-11e5-8edd-fdd255d783bd.png)
It's a work-around and it's summarized to this patch: http://pastebin.com/raw/hAVjxJYr, and since I don't really know C, I copy-pasted a lot from the source. 
The patch will create 3 options and can be found at org->nemo->icon-view: icon-width, custom-icon-width and custom-max-text-width.

**Comment by JosephMcc** [comment]
I would be okay with a bit less spacing as long as it wasn't too cramped but having so many options seems a bit over the top. If it's going to be changed, why not just change it and call it good?

**Comment by enriquezmark36** [comment]
Yeah, you're right. Too much options are confusing; it's also the reason I'm having a difficulty on plasma 5. I originally plan to add only 1 option as there was supposedly no "custom" options since its a bit complicated but will enable someone to manually tune the width even further.

**Comment by enriquezmark36** [comment]
I created a fork where the patch is applied, it's in here: https://github.com/enriquezmark36/nemo.
When does one create a pull request? I am a real noob in git.

**Comment by Vahan86** [comment]
@JosephMcc, can we mark this issue as a feature request?

**Comment by LogX7** [comment]
@enriquezmark36 where do i put the patch .. ?

**Comment by enriquezmark36** [comment]
@LogX7 you could check this pull request, https://github.com/linuxmint/nemo/pull/1095 and download all 3 commits?
[7bd14d30cabbce41041cde932c80d905fd57b8cc.patch](https://github.com/linuxmint/nemo/commit/7bd14d30cabbce41041cde932c80d905fd57b8cc.patch)
[8f62a2e9ebc32fe685e7a14c344e09be723a915f.patch](https://github.com/linuxmint/nemo/commit/8f62a2e9ebc32fe685e7a14c344e09be723a915f.patch)
[596d6499a4e3f5987b78d702beba5e75a19befa2.patch](https://github.com/linuxmint/nemo/commit/596d6499a4e3f5987b78d702beba5e75a19befa2.patch)
And, patch it in that order. I think you could do a 
`patch -p1 -i <path/to/patch>` on the source directory to apply it?
Really sorry, that this is about 3 years old so I have sort of forgotten how these patches work.

**Comment by LogX7** [comment]
@enriquezmark36 it asks for the file to patch ...

**Comment by enriquezmark36** [comment]
@LogX7 May I ask what is the output of the `patch` command along with the `ls .`  command? You may have not been running `patch` on the source directory.

**Comment by LogX7** [comment]
@enriquezmark36 here you go 
https://i.imgur.com/PP382qRg.png

**Comment by enriquezmark36** [comment]
@LogX7 Wow, it has been about like 2-3 years and nemo has already progressed by that much. ~Well, I'll try to port at least the first patch tomorrow.

**Comment by enriquezmark36** [comment]
Okay I think I may have successfully ported it. To my surprise they already have something that (almost) have the same effect as the second patch so it won't be needed. And the third patch can be easily merged to the first. And so, I am left with a single patch.

@LogX7 , please apply this commit instead. Same procedure as before.
[5023bb015aa852db89cc0d4e7d21c1359a1a0c7a.patch](https://github.com/enriquezmark36/nemo/commit/5023bb015aa852db89cc0d4e7d21c1359a1a0c7a.patch)

**Comment by LogX7** [comment]
wi'm doing 

     cd ~/Desktop 
     wget 5023bb015aa852db89cc0d4e7d21c1359a1a0c7a.patch
     patch -p1 -i 5023bb015aa852db89cc0d4e7d21c1359a1a0c7a.patch

it asks me for File to patch:

**Comment by enriquezmark36** [comment]
How? I already rebased it on the master branch? Sorry, for the trouble.
I have troubles with messaging again.

But first, @LogX7 , why are you on the ~/Desktop/ folder?
Have you extracted the source there as is?
I mean, is there that one folder that goes by the name
which starts with"nemo" created when
you extracted it on to your ~/Desktop?
If there is, `cd` to that directory and use patch again to the location of the 5023bb015aa852db89cc0d4e7d21c1359a1a0c7a.patch file.

Or, you could be using a version/revision aside from the master branch for
which I have no idea how or what it looked like.

If that's the case then,
@LogX7, could you try to clone/download the iconwidth branch of my [fork](https://github.com/enriquezmark36/nemo/tree/iconwidth)? It should be the same as the original master branch but with that commit added. 
You won't need to use the `patch` command there anymore. Just build/compile it as you should.

**Comment by gadass** [comment]
Hi @enriquezmark36, 

I'm writing only to thank you very much for your patch/fork of Nemo. It would be very nice if this feature was implemented as default in the future versions.

Cheers,

José Gadelha

**Comment by enriquezmark36** [comment]
@gadass , you're welcome. Seriously, I didn't thought of this being of use to others.

BTW, I updated the fork. Now, it exposes the setting in the preferences window.

**Comment by jretropie** [comment]
Does the patch still works?

**Comment by enriquezmark36** [comment]
@jretropie, I don't think that would apply anymore with the most recent nemo. It seems that they figured to scale the spacing, icon/thumbnail and text size with the zoom. It works so much better than the patch in my fork which just introduces a setting to manually set those sizes by force.

In simpler terms, this problem from about 4 years ago :) , mostly if not completely, has been solved. The patch should not work anymore.

**Comment by jretropie** [comment]
I still have spacing problem with icon on Nemo when I zoom in the max.

**Comment by serfer2** [comment]
Still having same issue

**Comment by p-kraszewski** [comment]
`nemo 3.8.5` on Ubuntu Budgie 19.04 - still the same issue. Space between icons is scaling faster than icons themselves. On a max zoom I have a single column of icons on 1280px monitor and two on 1920px one. 

![Zrzut ekranu z 2019-04-30 07-27-21](https://user-images.githubusercontent.com/1206223/56942187-8a4a7280-6b19-11e9-8a66-9baec1d10107.png)

**Comment by p-kraszewski** [comment]
> Still having same issue

I am also having this issue on a fresh Ubuntu Budgie 19.04, but it seems they use ancient nemo version 3.8.5. Check yours.

**Comment by fossfreedom** [comment]
@p-kraszewski enable UB's backports (budgie-welcome - recommendations) and then you will have nemo v4.0.6 available to upgrade to.
